edgelet/provisioning/src/provisioning.rs

```diff
@@ -207,31 +234,29 @@ impl Provision for ManualProvisioning {
     type Hsm = MemoryKeyStore;
 
     fn provision(
-        self,
+        &self,
         mut key_activator: Self::Hsm,
     ) -> Box<dyn Future<Item = ProvisioningResult, Error = Error> + Send> {
-        let ManualProvisioning {
-            key,
-            device_id,
-            hub,
-        } = self;
-
         info!(
             "Manually provisioning device \"{}\" in hub \"{}\"",
-            &device_id, &hub
+            self.device_id, self.hub
         );
         let result = key_activator
-            .activate_identity_key(KeyIdentity::Device, "primary".to_string(), key)
+            .activate_identity_key(KeyIdentity::Device, "primary".to_string(), self.key.clone())
             .map(|_| ProvisioningResult {
-                device_id,
-                hub_name: hub,
+                device_id: self.device_id.to_string(),
+                hub_name: self.hub.to_string(),
                 reconfigure: ReprovisioningStatus::DeviceDataNotUpdated,
                 sha256_thumbprint: None,
                 credentials: None,
             })
             .map_err(|err| Error::from(err.context(ErrorKind::Provision)));
         Box::new(result.into_future())
     }
+
+    fn reprovision(&self) -> Box<dyn Future<Item = (), Error = Error> + Send> {
+        Box::new(future::ok(()))
```

Comment:
Add a `// No reprovision action is needed for the manual provisioning mode.` comment like you have for DPS.